src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -318,10 +323,42 @@ object TaskBuilder {
         "MESOS_TASK_ID" -> taskId.map(_.getValue),
         "MARATHON_APP_ID" -> Some(app.id.toString),
         "MARATHON_APP_VERSION" -> Some(app.version.toString),
-        "MARATHON_APP_DOCKER_IMAGE" -> app.container.flatMap(_.docker.map(_.image))
+        "MARATHON_APP_DOCKER_IMAGE" -> app.container.flatMap(_.docker.map(_.image)),
+        "MARATHON_APP_RESOURCE_CPUS" -> Some(app.cpus.toString),
+        "MARATHON_APP_RESOURCE_MEM" -> Some(app.mem.toString),
+        "MARATHON_APP_RESOURCE_DISK" -> Some(app.disk.toString)
       ).collect {
           case (key, Some(value)) => key -> value
-        }.toMap
+        }.toMap ++ labelsToEnvVars(app.labels)
+    }
+  }
+
+  def labelsToEnvVars(labels: Map[String, String]): Map[String, String] = {
```

Comment:
Sorry to be picky: collect already filters automatically. May I suggest the following code:
```scala
def labelsToEnvVars(labels: Map[String, String]): Map[String, String] = {
  def escape(name: String): String = name.replaceAll("[^a-zA-Z0-9_]+", "_").toUpperCase
  val validLabels = labels.collect {
    case (key, value) if key.length<maxEnvLen && value.length<maxEnvLen => escape(key) -> value
  }

  val names = Map("MARATHON_APP_LABELS" -> validLabels.keys.mkString(" "))
  val values = validLabels.map { case (key, value) => s"$labelEnvironmentKeyPrefix$key" -> value }
  names ++ values
}
```